### ISSUE TITLE
Support for GOG.com Rayman Forever save/config location

### ISSUE THREAD

**Issue by Mystyle-48** [RayCarrot/RayCarrot.RCP.Metro#32]
As of 2019, the GOG.com version of Rayman Forever redirects save and configuration files for Rayman 1 and Designer to a subfolder of the game directory named `cloud_saves`, similar to other DOS games with GOG Galaxy cloud save support. RCP does not detect this alternate location, preventing configuration and progression from being properly viewed. May there be support for the `cloud_saves` location in RCP, similar to how the `%LOCALAPPDATA%\VirtualStore` redirect location is supported for Rayman 3's config file?

**Comment by RayCarrot** [comment]
The way which GOG Galaxy implements the save redirecting is sadly pretty flawed. It only does so for when the game is running in DOSBox, which is fine for the most part, but doesn't work for the map editor (Mapper.exe) since that's a standard Win32 executable. This causes there to be a de-sync between the files in the game, basically preventing you from creating your own levels.
I tried writing to GOG's support about this about a year ago or so, but I never got a reply and I suppose at this point they probably won't bother fixing it. Unless they have done so recently and I haven't noticed. I usually just recommend using the offline installer for Rayman Forever from GOG's website rather than the GOG Galaxy version since that one doesn't redirect the saves. But I suppose there are still people who will want to use GOG Galaxy, so it is a bit problematic that RCP doesn't take this into account yeah. I'll have a look at implementing an option for this, and like you suggested I could possibly reuse parts of the system I made for the VirtualStore redirecting that Windows does for some programs.

**Comment by Mystyle-48** [comment]
Yeah, the cloud save support is incomplete, and GOG is unlikely to bother to fix it for the mapper or Rayman By his Fans. However, the redirect also applies to the offline installer version, as the folder is still mounted by DOSBox itself (`dosboxRayman_single.conf`).

**Comment by RayCarrot** [comment]
Oh I see. I never noticed since I usually launch the game through RCP which doesn't use the GOG config file. Hm, that's a bit more problematic then yeah.
